python/ray/tune/sample.py

```diff
@@ -6,9 +6,64 @@
 
 import numpy as np
 
+# Backwards compatibility
+try:
+    # Added in numpy>=1.17 but we require numpy>=1.16
+    np_random_generator = np.random.Generator
+    LEGACY_RNG = False
+except AttributeError:
+
+    class np_random_generator:
+        pass
+
+    LEGACY_RNG = True
+
 logger = logging.getLogger(__name__)
 
 
+class _BackwardsCompatibleNumpyRng:
+    """Thin wrapper to ensure backwards compatibility between
+    new and old numpy randomness generators.
+    """
+    _rng = None
+
+    def __init__(
+            self,
+            generator_or_seed: Optional[Union[
+                "np_random_generator", np.random.RandomState, int]] = None):
+        if generator_or_seed is None:
+            self._rng = np.random
+        elif isinstance(generator_or_seed, np.random.RandomState):
+            self._rng = generator_or_seed
+        elif isinstance(generator_or_seed, np_random_generator):
+            self._rng = generator_or_seed
+        elif LEGACY_RNG:
+            self._rng = np.random.RandomState(generator_or_seed)
+        else:
+            self._rng = np.random.default_rng(generator_or_seed)
+
+    @property
+    def legacy_rng(self) -> bool:
+        return not isinstance(self._rng, np_random_generator)
+
+    @property
+    def rng(self):
+        return self._rng if self._rng else np.random
```

Comment:
We don't need this anymore, right?

```suggestion
    @property
    def rng(self):
        return self._rng
```

or remove

Reply:
We do need it, it's the default behavior

Reply:
Oh right, this is actually necessary as setting `self._rng = np.random` causes pickling issues